src/lib/taxaFetchAPI.js

```diff
@@ -0,0 +1,43 @@
+// @flow
+
+import inatjs from "inaturalistjs";
+
+const ANCESTOR_FIELDS = {
+  name: true,
+  preferred_common_name: true,
+  rank: true
+};
+
+const PHOTO_FIELDS = {
+  id: true,
+  attribution: true,
+  license_code: true,
+  url: true
+};
+
+const FIELDS = {
+  ancestors: ANCESTOR_FIELDS,
+  name: true,
+  preferred_common_name: true,
+  rank: true,
+  taxon_photos: {
+    photo: PHOTO_FIELDS
+  },
+  wikipedia_summary: true,
+  wikipedia_url: true
+};
+
+const fetchTaxa = async ( id: number ): Promise<any> => {
+  const params = {
+    fields: FIELDS
+  };
+
+  try {
+    const { results } = await inatjs.taxa.fetch( id, params );
+    return results[0];
+  } catch ( e ) {
+    throw new Error( JSON.stringify( e.response ) );
```

Comment:
Not sure this is adding anything. inatjs will throw an error if something goes wrong internally or if the request fails for some reason, which will satisfy React Query's need for an error. If we want to perform some standardized error handling this is just percolating up an error with a stringified object, and it's going to be tricky for whatever code is dealing with this error to parse out meaningful information.

Something like this might be preferable:

```javascript
if ( !e.response ) { throw e; }
const errorJson = await e.response.json( );
const error = new Error( errorJson.status );
throw error;
```

That way

1. If it's *not* an HTTP error we just throw it in its original state, and the consuming code can handle it as it sees fit
2. If it is an HTTP error we parse the meaningful JSON response and the HTTP status and return then as attributes of the error

I'm not convinced that's superior to just throwing the original error all the time, but it does help separate concerns. We'd want to extract it so it could be shared across these API files, though.